### PR TITLE
Multimodal Support

### LINKED ISSUE #242: Add multimodal support in prompt_template for easier prompting.
### Description

First thing that comes to mind is to add something like this
```python
prompt_template = """
SYSTEM: 
...

USER:
...

IMAGE:
{image_url}
```

### PR DESCRIPTION
**Pull request by willbakst**
Closes #242 

Added multimodal support for Anthropic, Gemini, OpenAI (and LiteLLM for providers that support it).

Cohere, Groq, and Mistral don't support multimodal (at least to my knowledge), so I've thrown errors when attempting to use multimodal prompt templating with those providers.

The multimodal prompt template parser currently handles text, image, and images (which is just a list of images). It is implemented such that it should be able to be easily extended to additional modalities such as audio, video, etc.

### CODE REVIEW

File: mirascope/core/base/_utils/parse_content_template.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    template: str
+    type: _PartType
+
+
+class _ImageOptions(TypedDict, total=False):
+    detail: _Detail
+
+
+def _parse_parts(template: str) -> List[_Part]:
+    pattern = r"\{([^:{}]+):([^{}]+)\}"
```
Comment (brenkao) [inline mirascope/core/base/_utils/parse_content_template.py:24]: short docstring to describe whats happening, especially in regards to the splitting

File: mirascope/core/base/_utils/parse_content_template.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+"""This module provides a function to parse content parts from a prompt template."""
+
+import imghdr
```
Comment (brenkao) [inline mirascope/core/base/_utils/parse_content_template.py:3]: looks like this is getting deprecated in python 3.13. either implement the alternative or put a note https://docs.python.org/3/library/imghdr.html

**Review by brenkao — APPROVED** [review]

File: mirascope/core/base/_utils/parse_content_template.py (hunk 1 above)
Comment (willbakst) [reply mirascope/core/base/_utils/parse_content_template.py:24]: Done.

File: mirascope/core/base/_utils/parse_content_template.py (hunk 2 above)
Comment (willbakst) [reply mirascope/core/base/_utils/parse_content_template.py:3]: Done. I wrote our own detection function and also added handling for each provider to throw an error if the user provides an image type that provider doesn't support.

**Pull request merged**